flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/service/FlinkService.java

```diff
@@ -192,7 +194,8 @@ private JarRunResponseBody runJar(
                             job.getParallelism() > 0 ? job.getParallelism() : null,
                             jobID,
                             job.getAllowNonRestoredState(),
-                            savepoint);
+                            savepoint,
+                            RestoreMode.DEFAULT);
```

Comment:
Based on the CI we should only set the RestoreMode if version is newer than 1.14 otherwise set null